### PR TITLE
SEED-485 redeploy webhooks

### PR DESCRIPTION
**Pull request by jerith**

### CODE REVIEW

**Comment by jerith** [comment]
I'm not really sure what to do for the UI here. We probably need to display the webhook URL somewhere and provide a good way to set or regenerate the token.

File: mc2/controllers/base/models.py [hunk 1]
```diff
@@ -54,6 +54,8 @@ class Controller(PolymorphicModel):
             'is automatically set each time the item is saved.')
     )
 
+    webhook_token = models.UUIDField(null=True)
```
Comment (smn) [inline mc2/controllers/base/models.py]: Can we pregenerate a UUID4? Do we want to do that in the UI or on the model level?

Comment (jerith) [inline mc2/controllers/base/models.py]: We probably could, but I have a mild preference for not exposing the webhook unless it's going to be used.

Comment (smn) [inline mc2/controllers/base/models.py]: So in the UI it is.

**Comment by smn** [comment]
:+1:

File: mc2/controllers/base/views.py [hunk 2]
```diff
@@ -192,3 +193,27 @@ def post(self, request, controller_pk):
...
+    Technically, this is a 'URL capability'.
+    """
+
+    def get(self, request, controller_pk, token):
+        return HttpResponseNotAllowed(['POST'])
+
+    def post(self, request, controller_pk, token):
+        controller = get_object_or_404(Controller, pk=controller_pk)
+        if str(controller.webhook_token) != token:
+            return HttpResponseNotFound()
```
Comment (miltontony) [inline mc2/controllers/base/views.py]: Why not `forbidden`

Comment (jerith) [inline mc2/controllers/base/views.py]: `403 Forbidden` means "this thing exists but you're not authorised to have it", `404 Not Found` means "this thing doesn't exist". The client is making a request to `http://example.com/foo/bar/token/` without auth, so `404` is the more appropriate response if the token doesn't match.

**Comment by miltontony** [comment]
:+1:

**Comment by smn** [comment]
I remember now what we're missing here, the URL needs to be CSRF exempt.

**Pull request merged**